### ISSUE TITLE
[V3 Downloader] Impossible to use the `[p]repo add` command

### ISSUE THREAD

**Issue by laggron42** [Cog-Creators/Red-DiscordBot#1226]
### Type:

- [ ] Suggestion
- [x] Bug

### Brief description of the problem

Impossible to use `[p]repo add` on V3

### Actual behavior

Doing the normal command

### Steps to reproduce

1. `[p]repo add discord_cogs https://github.com/Malarne/discord_cogs`
![screen shot 2017-12-31 at 16 36 24](https://user-images.githubusercontent.com/23153234/34462649-c97e86e8-ee48-11e7-8f85-40d5339bc51f.png)

### Error

```
[31/12/2017 16:30] ERROR downloader _repo_add 218: Something went wrong during the cloning process.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/downloader.py", line 212, in _repo_add
    branch=branch
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/repo_manager.py", line 540, in add_repo
    await r.clone()
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/repo_manager.py", line 198, in clone
    raise CloningError("Error when running git clone.")
downloader.errors.CloningError: Error when running git clone.
```
Unfortunately unable to give more details to the *not clear* error, as I don't know how to.

**Comment by laggron42** [comment]
Sorry, I didn't post all the error. It should be easier to understand now
```
[31/12/2017 16:30] ERROR downloader _repo_add 218: Something went wrong during the cloning process.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/downloader.py", line 212, in _repo_add
    branch=branch
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/repo_manager.py", line 540, in add_repo
    await r.clone()
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/repo_manager.py", line 198, in clone
    raise CloningError("Error when running git clone.")
downloader.errors.CloningError: Error when running git clone.
fatal: Too many arguments.

usage: git clone [<options>] [--] <repo> [<dir>]

    -v, --verbose         be more verbose
    -q, --quiet           be more quiet
    --progress            force progress reporting
    -n, --no-checkout     don't create a checkout
    --bare                create a bare repository
    --mirror              create a mirror repository (implies bare)
    -l, --local           to clone from a local repository
    --no-hardlinks        don't use local hardlinks, always copy
    -s, --shared          setup as shared repository
    --recurse-submodules[=<pathspec>]
                          initialize submodules in the clone
    -j, --jobs <n>        number of submodules cloned in parallel
    --template <template-directory>
                          directory from which templates will be used
    --reference <repo>    reference repository
    --reference-if-able <repo>
                          reference repository
    --dissociate          use --reference only while cloning
    -o, --origin <name>   use <name> instead of 'origin' to track upstream
    -b, --branch <branch>
                          checkout <branch> instead of the remote's HEAD
    -u, --upload-pack <path>
                          path to git-upload-pack on the remote
    --depth <depth>       create a shallow clone of that depth
    --shallow-since <time>
                          create a shallow clone since a specific time
    --shallow-exclude <revision>
                          deepen history of shallow clone, excluding rev
    --single-branch       clone only one branch, HEAD or --branch
    --no-tags             don't clone any tags, and make later fetches not to follow them
    --shallow-submodules  any cloned submodules will be shallow
    --separate-git-dir <gitdir>
                          separate git dir from working tree
    -c, --config <key=value>
                          set config inside the new repository
    -4, --ipv4            use IPv4 addresses only
    -6, --ipv6            use IPv6 addresses only

[31/12/2017 16:41] ERROR downloader _repo_add 218: Something went wrong during the cloning process.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/downloader.py", line 212, in _repo_add
    branch=branch
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/repo_manager.py", line 540, in add_repo
    await r.clone()
  File "/usr/local/lib/python3.6/site-packages/redbot/cogs/downloader/repo_manager.py", line 198, in clone
    raise CloningError("Error when running git clone.")
downloader.errors.CloningError: Error when running git clone.
```

**Comment by aikaterna** [comment]
Seems like downloader doesn't like special characters in the repo name. I tried adding the repo without the underscore in the name and it worked okay and I was able to add a cog from the repo.

**Comment by tekulvw** [comment]
Fixed with #1349
